### PR TITLE
[Merged by Bors] - feat: port Analysis.Calculus.Deriv.Mul

### PR DESCRIPTION
**Pull request by urkud**
---

- [x] depends on: #4434

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by semorrison** [comment]
This PR/issue depends on:
* leanprover-community/mathlib4#4434
By **[Dependent Issues](https://github.com/z0al/dependent-issues)** (🤖). Happy coding!

**Comment by semorrison** [comment]
This PR/issue depends on:
* ~~leanprover-community/mathlib4#4434~~
By **[Dependent Issues](https://github.com/z0al/dependent-issues)** (🤖). Happy coding!

**Comment by ChrisHughes24** [comment]
bors merge

**Comment by urkud** [comment]
Avoiding merge conflicts on the queue:
bors r-

File: Mathlib/Analysis/Calculus/Deriv/Mul.lean [hunk 1]
```diff
@@ -0,0 +1,417 @@
...
+import Mathlib.Analysis.Calculus.FDeriv.Mul
+import Mathlib.Analysis.Calculus.FDeriv.Add
+
+/-!
+# Derivative of `f x * g x`
+
+In this file we prove formulas for `(f x * g x)'` and `(f x • g x)'`.
+
+For a more detailed overview of one-dimensional derivatives in mathlib, see the module docstring of
+`analysis/calculus/deriv/basic`.
```
Comment (fpvandoorn) [inline Mathlib/Analysis/Calculus/Deriv/Mul.lean:21]:
```suggestion
`Analysis/Calculus/Deriv/Basic`.
```
also in other PRs

**Comment by urkud** [comment]
bors merge

**Pull request closed without merging**